### PR TITLE
Improve alignment of `latest-tag-button` and `list-prs-for-file`

### LINKED ISSUE #4229: `latest-tag-button` and `list-prs-for-file` are not aligned to the branch selector
1. https://github.com/sindresorhus/refined-github/blob/48dcf91bc6d20e5fde93af139636a3a7cef0e412/source/github-helpers/index.ts 
2. Resize to 800px

Screen Shot 4

### PR DESCRIPTION
**Pull request by cheap-glitch**
Fix #4229 

## Test URLs

* [Latest tag + 3 PRs](https://github.com/sindresorhus/refined-github/blob/main/readme.md)
* [Latest tag + a single PR](https://github.com/sindresorhus/refined-github/blob/48dcf91bc6d20e5fde93af139636a3a7cef0e412/source/github-helpers/index.ts)
* [Just a tag](https://github.com/sindresorhus/refined-github/blob/main/source/features/ci-link.tsx)
* [Repo tree](https://github.com/sindresorhus/refined-github/tree/main/distribution)
* [Repo root](https://github.com/sindresorhus/refined-github)

This PR should be tested both with `latest-tag-button` and `list-prs-for-file` enabled separately *and* enabled together.

## Screenshots

_**Full width (no changes):**_
image

_**Wrapped breadcrumb + 2 action buttons:**_
image

_**Mobile version (wrapped breadcrumb + single button):**_
image

This PR:
 * puts the latest tag button and the PRs dropdown in the wrapper around the branches selection dropdown
 * wraps this wrapper + the breadcrumb in another flex div to make it wrap without affecting the buttons on the right

### CODE REVIEW

File: source/features/latest-tag-button.tsx [hunk 1]
```diff
@@ -90,13 +90,11 @@ async function init(): Promise<false | void> {
 	});
 
 	const link = (
-		<a className="btn btn-sm btn-outline ml-2 flex-self-center css-truncate rgh-latest-tag-button" href={String(url)}>
+		<a className="btn btn-sm btn-outline ml-0 mr-2 flex-self-center v-align-top css-truncate rgh-latest-tag-button" href={String(url)}>
```
Comment (cheap-glitch) [inline source/features/latest-tag-button.tsx:93]: `.v-align-top` is needed to cancel the `vertical-align: middle` that comes with `.btn` and causes a misalignment when both features are under the branches dropdown (like in the second screenshot).

File: source/github-helpers/add-button-next-to-branch-select-menu.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import select from 'select-dom';
+import elementReady from 'element-ready';
+
+export default async function addButtonNextToBranchSelectMenu(button: Element): Promise<void> {
```
Comment (cheap-glitch) [inline source/github-helpers/add-button-next-to-branch-select-menu.ts:4]: Suggestions for a shorter name welcomed :smile:

Comment (yakov116) [reply source/github-helpers/add-button-next-to-branch-select-menu.ts:4]: appendToBranchMenu ?

Comment (yakov116) [inline source/github-helpers/add-button-next-to-branch-select-menu.ts:4]: I don't think we need a new file. Export it from one of the other feature files.

Comment (fregante) [reply source/github-helpers/add-button-next-to-branch-select-menu.ts:4]: addAfterBranchSelector.

File: source/features/latest-tag-button.tsx [hunk 3]
```diff
@@ -1,5 +1,7 @@
 import './latest-tag-button.css';
 import React from 'dom-chef';
+import select from 'select-dom';
+import elementReady from 'element-ready';
 import cache from 'webext-storage-cache';
 import * as pageDetect from 'github-url-detection';
```
Comment (yakov116) [inline source/features/latest-tag-button.tsx]:
```suggestion
import cache from 'webext-storage-cache';
import select from 'select-dom';
import elementReady from 'element-ready';
import * as pageDetect from 'github-url-detection';
```

File: source/features/latest-tag-button.tsx [hunk 4]
```diff
@@ -13,6 +14,20 @@ import {groupButtons} from '../github-helpers/group-buttons';
 import getDefaultBranch from '../github-helpers/get-default-branch';
 import {buildRepoURL, getCurrentCommittish, getLatestVersionTag, getRepo} from '../github-helpers';
 
+export default async function addAfterBranchSelector(button: Element): Promise<void> {
```
Comment (yakov116) [inline source/features/latest-tag-button.tsx:17]:
```suggestion
// eslist-disable......
export async function addAfterBranchSelector(button: Element): Promise<void> {
```
Nitpick
We dont do defaults from feature files

File: source/features/latest-tag-button.tsx [hunk 5]
```diff
@@ -13,6 +14,21 @@ import {groupButtons} from '../github-helpers/group-buttons';
...
 
+// eslint-disable-next-line import/prefer-default-export
+export async function addAfterBranchSelector(button: Element): Promise<void> {
+	const branchSelector = (await elementReady('#branch-select-menu', {waitForChildren: false}))!;
+	const wrapper = branchSelector.closest('.position-relative')!;
+	button.classList.add('mt-md-2', 'mt-lg-0');
+	wrapper.append(button);
+	if (wrapper.classList.contains('rgh-button-added')) {
+		return;
+	}
```
Comment (fregante) [inline source/features/latest-tag-button.tsx:25]: Seems unnecessary, they all add classes afterwards, it's not a huge problem to run them again

**Comment by cheap-glitch** [comment]
This isn't great: (814px width):
image

Even worse (774px) :face_with_head_bandage: 
image

Maybe this can be fixed by giving the flex item wrapping the buttons a higher "grow factor".

**Comment by fregante** [comment]
> This isn't great: (814px width):

Maybe they should wrap at a lower resolution, e.g. when the path starts wrapping on its own. I don't know if this can be done via flex-grow/shrink without using pre-set media queries

**Comment by fregante** [comment]
I like this: Put the path on the second line when it doesn't fit




This is smart. Let's try to replicate it automatically via flex, without using media queries.

File: source/features/latest-tag-button.tsx [hunk 6]
```diff
@@ -12,6 +13,25 @@ import GitHubURL from '../github-helpers/github-url';
 import {groupButtons} from '../github-helpers/group-buttons';
 import getDefaultBranch from '../github-helpers/get-default-branch';
 import {buildRepoURL, getCurrentCommittish, getLatestVersionTag, getRepo} from '../github-helpers';
+import {wrapAll} from '../helpers/dom-utils';
```
Comment (yakov116) [inline source/features/latest-tag-button.tsx:16]:
```suggestion
import {wrapAll} from '../helpers/dom-utils';
import {groupButtons} from '../github-helpers/group-buttons';
import getDefaultBranch from '../github-helpers/get-default-branch';
import {buildRepoURL, getCurrentCommittish, getLatestVersionTag, getRepo} from '../github-helpers';
```

**Comment by fregante** [comment]
Thank you! It seems to work well, however the alignment could be improved, can it be done?

## Native

Screen Shot 2

## This PR

Screen Shot 3

**Comment by fregante** [comment]
This PR now

Screen Shot 1
Screen Shot

**Comment by fregante** [comment]
The margins look good

File: source/features/latest-tag-button.tsx [hunk 7]
```diff
@@ -17,9 +17,14 @@ import {buildRepoURL, getCurrentCommittish, getLatestVersionTag, getRepo} from '
 
 // eslint-disable-next-line import/prefer-default-export
 export async function addAfterBranchSelector(button: Element): Promise<void> {
+	button.classList.add('ml-2');
 	const branchSelector = (await elementReady('#branch-select-menu', {waitForChildren: false}))!;
 	const branchSelectorWrapper = branchSelector.closest('.position-relative')!;
-	button.classList.add('ml-2');
+	if (!select.exists('.breadcrumb')) {
```
Comment (fregante) [inline source/features/latest-tag-button.tsx:23]: Breadcrumbs are selected twice now, we can reuse a single select

**Comment by fregante** [comment]
# 🙌 

Perfect!

**Comment by fregante** [comment]
Thanks for all the test URLs and screenshots, no doubt they'll be useful in the future

**Comment by cheap-glitch** [comment]
> Thanks for all the test URLs

Sadly, most of them are outdated now... I wonder if we should make an "official" RGH dummy test repo, for when we need specific test URLs like these 🤔

**Comment by fregante** [comment]
> no doubt they'll be useful in the future

They are

**Pull request merged**